File: domain/src/main/java/com/ahn/domain/currency/usecase/SaveExchangeSelectionUseCase.kt
package com.ahn.domain.currency.usecase

import com.ahn.domain.currency.repository.CurrencySelectionRepository
import javax.inject.Inject

class SaveExchangeSelectionUseCase @Inject constructor(private val repository: CurrencySelectionRepository) {
    suspend operator fun invoke(fromCode: String, toCode: String) {
        repository.saveExchangeSelection(fromCode, toCode)
    }
}

File: ios/Packages/Domain/Sources/Domain/Currency/UseCase/SaveExchangeSelectionUseCase.swift
public struct SaveExchangeSelectionUseCase: Sendable {
    private let repository: any CurrencySelectionRepository

    public init(repository: any CurrencySelectionRepository) {
        self.repository = repository
    }

    public func callAsFunction(fromCode: String, toCode: String) async throws {
        try await repository.saveExchangeSelection(fromCode: fromCode, toCode: toCode)
    }
}
